src/state.py

```diff
@@ -873,3 +881,41 @@ def _parse_script_url(charm: ops.CharmBase) -> str | None:
     if not parsed_url.scheme or not parsed_url.hostname:
         raise InvalidScriptURLError("Invalid script URL, must contain scheme and hostname.")
     return script_url_str
+
+
+class InvalidSecretError(CharmConfigInvalidError):
+    """Represents an error when fetching secrets."""
+
+
+def _parse_script_secrets(charm: ops.CharmBase) -> dict[str, str] | None:
+    """Parse secrets to load as environment variables for the external script.
+
+    Args:
+        charm: The running charm instance.
+
+    Raises:
+        InvalidSecretError: If a secret of invalid format (secrets separated by space)
+    """
+    script_secret_id = typing.cast(str, charm.config.get(SCRIPT_SECRET_ID_CONFIG_NAME, ""))
+    script_secret = typing.cast(str, charm.config.get(SCRIPT_SECRET_CONFIG_NAME, ""))
```

Comment:
Could we also block the charm if the config is set and running on a version of juju that supports user secrets?

Reply:
I think I may need Juju team's help on this - I'm not sure about the capabilities of Juju within each versions.
Some juju versions user secrets cannot be set, some don't support secrets at all (which I think we shouldn't cover for such old juju versions though).
So either we cater for such compatibility matrix and start handling it in the charm by blocking it or we could just let it error out.

Reply:
I think there is a way to check the juju version and just block if the version of juju is too low?

Reply:
If that is the case, i think the charm can use min_juju_version in the matrix and just not let deployment rather than blocking it.

Reply:
Juju 3.3 introduced user secrets, so just block if we're on that version or later: https://discourse.charmhub.io/t/roadmap-releases/5064

Reply:
Using the assumes matrix.

Reply:
But we want to support a config option on versions of Juju 3.2 or earlier and user secrets on version 3.3 and later? Not block the deployment?

Reply:
Would you know an example of how you might be able to fetch the Juju version from the ops library?

Reply:
I think `paas-charm` has an implementation for it, let me check

Reply:
Here you go: https://github.com/canonical/paas-charm/blob/b795408df3f8b6c1babbee89ba757be1e6de8f3d/src/paas_charm/utils.py#L47